messaging/src/main/java/org/axonframework/commandhandling/distributed/DistributedCommandBus.java

```diff
@@ -177,25 +158,26 @@ public <C, R> void dispatch(@Nonnull CommandMessage<C> command,
         CommandMessage<? extends C> interceptedCommand = intercept(command);
         MessageMonitor.MonitorCallback messageMonitorCallback = messageMonitor.onMessageIngested(interceptedCommand);
         Optional<Member> optionalDestination = commandRouter.findDestination(interceptedCommand);
-        if (optionalDestination.isPresent()) {
-            Member destination = optionalDestination.get();
-            try {
+        Span span = spanFactory.createDispatchSpan(() -> "DistributedCommandBus.dispatch", command).start();
+        try (SpanScope ignored = span.makeCurrent()) {
+            if (optionalDestination.isPresent()) {
+                Member destination = optionalDestination.get();
                 connector.send(destination,
                                interceptedCommand,
                                new MonitorAwareCallback<>(callback, messageMonitorCallback));
-            } catch (Exception e) {
-                messageMonitorCallback.reportFailure(e);
-                destination.suspect();
-                callback.onResult(interceptedCommand, asCommandResultMessage(
-                        new CommandDispatchException(DISPATCH_ERROR_MESSAGE + ": " + e.getMessage(), e)
-                ));
+                span.end();
+            } else {
+                throw new NoHandlerForCommandException(
+                        format("No node known to accept command [%s].", interceptedCommand.getCommandName())
+                );
             }
-        } else {
-            NoHandlerForCommandException exception = new NoHandlerForCommandException(
-                    format("No node known to accept command [%s].", interceptedCommand.getCommandName())
-            );
-            messageMonitorCallback.reportFailure(exception);
-            callback.onResult(interceptedCommand, asCommandResultMessage(exception));
+        } catch (Exception e) {
+            span.recordException(e).end();
```

Comment:
So, this end is fine. It might be best to move the `end()` invocation to a `finally`, and remove it both here and up there.

Reply:
Thanks, moved it to the finalyl block.